test/e2e/deployment.go

```diff
@@ -175,9 +201,28 @@ func testRollingUpdateDeployment(f *Framework) {
 
 	err = waitForDeploymentStatus(c, ns, deploymentName, replicas, replicas-1, replicas+1, 0)
 	Expect(err).NotTo(HaveOccurred())
+
+	// Verify that the pods were scaled up and down as expected. We use events to verify that.
+	deployment, err := c.Deployments(ns).Get(deploymentName)
+	Expect(err).NotTo(HaveOccurred())
+	waitForEvents(c, ns, deployment, 4)
+	events, err := c.Events(ns).Search(deployment)
+	if err != nil {
+		Logf("error in listing events: %s", err)
+		Expect(err).NotTo(HaveOccurred())
+	}
+	// There should be 4 events, 2 to scale up the new RC and 2 to scale down the old RC.
+	Expect(len(events.Items)).Should(Equal(4))
+	newRC, err := deploymentutil.GetNewRC(*deployment, c)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(newRC).NotTo(Equal(nil))
+	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up rc %s to 1", newRC.Name)))
+	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down rc %s to 1", rcName)))
+	Expect(events.Items[2].Message).Should(Equal(fmt.Sprintf("Scaled up rc %s to 3", newRC.Name)))
+	Expect(events.Items[3].Message).Should(Equal(fmt.Sprintf("Scaled down rc %s to 0", rcName)))
 }
 
-func testRollingUpdateDeploymentEvents(f *Framework) {
```

Comment:
I will suggest leaving testRollingUpdateDeployment and testRollingUpdateDeploymentEvents as is and just adding testRecreateDeployment